WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -1622,6 +1634,100 @@ private static async Task FinishBranchActionAsync(int queueId, DataRow dataRowWi
+                return result;
+            }
+
+            // If the database to be deleted is the same as the currently connected database, then it's most likely the main branch, so stop the deletion.
+            if (databaseConnection.ConnectedDatabase == settings.DatabaseName)
+            {
+                await logService.LogError(logger, LogScopes.RunBody, branchQueue.LogSettings, $"Trying to delete a branch, but it looks to be the main branch and for safety reasons we don't allow the main branch to be deleted. Queue ID was: {queueId}", configurationServiceName, branchQueue.TimeId, branchQueue.Order);
+
+                error = "Trying to delete a branch, but it looks to be the main branch and for safety reasons we don't allow the main branch to be deleted.";
+                result.Add("ErrorMessage", error);
+                result.Add("Success", false);
+                databaseConnection.AddParameter("now", DateTime.Now);
+                databaseConnection.AddParameter("error", error);
+                await databaseConnection.ExecuteAsync($"UPDATE {WiserTableNames.WiserBranchesQueue} SET finished_on = ?now, success = 0, errors = ?error WHERE id = ?queueId");
+                return result;
+            }
+
+            // The name of the branch database is always prefixed with the name of the main branch database.
+            // If that is not the case, then we're trying to delete a branch of a different tenant, so stop the deletion.
+            if (!settings.DatabaseName.StartsWith(databaseConnection.ConnectedDatabase))
+            {
+                await logService.LogError(logger, LogScopes.RunBody, branchQueue.LogSettings, $"Trying to delete a branch, but it looks like this branch does not belong to the customer that this WTS is connected to. Queue ID was: {queueId}", configurationServiceName, branchQueue.TimeId, branchQueue.Order);
+
+                error = "Trying to delete a branch, but it looks like this branch does not belong to the customer that this WTS is connected to.";
+                result.Add("ErrorMessage", error);
+                result.Add("Success", false);
+                databaseConnection.AddParameter("now", DateTime.Now);
+                databaseConnection.AddParameter("error", error);
+                await databaseConnection.ExecuteAsync($"UPDATE {WiserTableNames.WiserBranchesQueue} SET finished_on = ?now, success = 0, errors = ?error WHERE id = ?queueId");
+                return result;
+            }
+
+            try
+            {
+                // Delete the database, if it exists.
+                if (await databaseHelpersService.DatabaseExistsAsync(settings.DatabaseName))
+                {
+                    await databaseHelpersService.DropDatabaseAsync(settings.DatabaseName);
+                }
+            }
+            catch (Exception exception)
+            {
+                logger.LogError(exception, $"Failed to delete the branch '{settings.DatabaseName}'.");
```

Comment:
Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.